tests/unit/plugins/test_authority.py

```diff
@@ -0,0 +1,148 @@
+    session = builder.session
+    with contextlib.closing(session):
+        gce_authority = builder.build_authority()
+
+    auth_client_calls = [
+        mocker.call(
+            authority_config['keyfile'],
+            scopes=authority_config['scopes'],
+            session=session),
+    ] * 2
+
+    auth_client.assert_has_calls(auth_client_calls, any_order=True)
+    expected_metadata_blacklist = [
+        dict([pair]) for pair in authority_config['metadata_blacklist']
+    ]
+    gce_client.assert_called_once_with(
+        auth_client(),
+        session,
+        blacklisted_tags=authority_config['tag_blacklist'],
+        blacklisted_metadata=expected_metadata_blacklist)
+
+    assert crm_client() is gce_authority.crm_client
+    assert gce_client() is gce_authority.gce_client
+    assert rrset_channel is gce_authority.rrset_channel
+    assert authority_config is gce_authority.config
+
+
+@pytest.mark.asyncio
+async def test_builder_error_on_required_config(authority_config, caplog):
+    caplog.set_level(logging.ERROR)
+    del authority_config['keyfile']
+    rrset_channel = asyncio.Queue()
+    kwargs = {}
+
+    builder = authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with pytest.raises(exceptions.GCPConfigError), contextlib.closing(session):
+        builder.build_authority()
+    assert 1 == len(caplog.records)
+
+
+@pytest.mark.asyncio
+async def test_start_publishes_msg_to_channel(mocker, authority_config,
+                                              get_gce_client, create_mock_coro):
+    instance_data = [{
+        'hostname': f'host-{i}',
+        'internal_ip': f'192.168.1.{i}',
+        'external_ip': f'1.1.1.{i}'
+    } for i in range(1, 4)]
+    active_projects_mock, active_projects_coro = create_mock_coro()
+    active_projects_mock.return_value = [{'projectId': 1}, {'projectId': 2}]
+    crm_client = get_gce_client(gcrm.GCRMClient)
+    crm_client.list_all_active_projects = active_projects_coro
+
+    list_instances_mock, list_instances_coro = create_mock_coro()
+    list_instances_mock.return_value = instance_data
+    gce_client = get_gce_client(gce.GCEClient)
+    gce_client.list_instances = list_instances_coro
+
+    rrset_channel = asyncio.Queue()
+    gce_authority = authority.GCEAuthority(authority_config, crm_client,
+                                           gce_client, rrset_channel)
+
+    await gce_authority.start()
+
+    expected_rrsets = [{
+        'name': record['hostname'],
+        'type': 'A',
+        'rrdatas': [record['external_ip']]
+    } for record in instance_data] * 2
+    expected_msg = {
+        'zone': authority_config['zone'],
+        'rrsets': expected_rrsets
+    }
+
+    # one message includes data for all projects
+    assert expected_msg == (await rrset_channel.get())
+    # start also calls self.done at the end
+    assert None is (await rrset_channel.get())
```

Comment:
`assert not (await rrset_channel.get())`?

Reply:
I'd prefer to check specifically for `None` since the plugins check for `None` and not for falsy things, too:

```python
# reconciler.py
desired_rrset = await self.rrset_channel.get()
if desired_rrset is None:
```